src/ucp/core/ucp_ep.c

```diff
@@ -214,10 +214,12 @@ static void ucp_ep_delete(ucp_ep_h ep)
     ucs_callbackq_remove_if(&ep->worker->uct->progress_q,
                             ucp_wireup_msg_ack_cb_pred, ep);
     if (!(ep->flags & UCP_EP_FLAG_INTERNAL)) {
-        ucp_worker_keepalive_remove_ep(ep);
+        if (ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL) {
+            ucp_worker_keepalive_remove_ep(ep);
+        }
+        ucs_list_del(&ucp_ep_ext_gen(ep)->ep_list);
```

Comment:
why moved?

Reply:
since no need to remove memtype EP from worker's all_ep list, since it's not added there

Reply:
weird: if we didn't add it, the list_del would fail

Reply:
> weird: if we didn't add it, the list_del would fail

looks like we just don't catch it
we create mem_type EP by means of:
https://github.com/openucx/ucx/blob/10f043c1876f99bfcfd54c5dd3c62d5c3d142dce/src/ucp/core/ucp_ep.c#L169
which adds EP to the worker's all_eps list if the following is not a true:
https://github.com/openucx/ucx/blob/10f043c1876f99bfcfd54c5dd3c62d5c3d142dce/src/ucp/core/ucp_ep.c#L195
and then we destroy mem_type EPs here:
https://github.com/openucx/ucx/blob/10f043c1876f99bfcfd54c5dd3c62d5c3d142dce/src/ucp/core/ucp_ep.c#L331
which calls for each mem_type EP:
https://github.com/openucx/ucx/blob/10f043c1876f99bfcfd54c5dd3c62d5c3d142dce/src/ucp/core/ucp_ep.c#L346

Reply:
ok.. really weird we don't see failures

Reply:
> ok.. really weird we don't see failures

yes, I also wonder "why?"

Reply:
looks like it's because of https://github.com/openucx/ucx/blob/f7d2aa3ceca0dc7f91f933ba08973dbcd0be3bb1/src/ucp/core/ucp_ep.c#L145

Reply:
yes, good catch
anyway I'd leave code without deleting EP from the list if it's not really added there

Reply:
sure, we must not delete ep from the list, it just worked because ext_gen(ep)->ep_list was (wrongfully) initialized as list head

Reply:
> sure, we must not delete INTERNAL ep from the list, it just worked because ext_gen(ep)->ep_list was (wrongfully) initialized as list head

should we delete it in a separate PR?

Reply:
yes